api/src/main/java/io/druid/data/input/FirehoseFactory.java

```diff
@@ -36,7 +37,13 @@
    * If this method returns null, then any attempt to call hasMore(), nextRow(), commit() and close() on the return
    * value will throw a surprising NPE.   Throwing IOException on connection failure or runtime exception on
    * invalid configuration is preferred over returning null.
+   * <p/>
+   * Some fire hoses like {@link PrefetchableTextFilesFirehoseFactory} may use a temporary
+   * directory to cache data in it.
+   *
+   * @param parser             an input row parser
+   * @param temporaryDirectory a directory where temporary files are stored
    */
-  public Firehose connect(T parser) throws IOException, ParseException;
+  Firehose connect(T parser, File temporaryDirectory) throws IOException, ParseException;
```

Comment:
Would you tell me what kind of temp dir management you think? I think the temp dir should be managed by the caller of PrefetchableFirehoseFactory.connect() because PrefetchableFirehoseFactory cannot decide when it removes the temp dir by itself.